### PR TITLE
Add commandlineArguments to XcodeGenKit Scheme specifications

### PR DESCRIPTION
**Pull request by rahul-malik**
These are the necessary changes after this PR in `xcproj` lands:
https://github.com/xcodeswift/xcproj/pull/167

### CODE REVIEW

**Comment by yonaskolb** [comment]
@rahul-malik I’d rather not merge anything that doesn’t point to a specific depndency version

File: Sources/ProjectSpec/Target.swift [hunk 1]
```diff
@@ -130,11 +130,13 @@ public struct TargetScheme {
     public var testTargets: [String]
     public var configVariants: [String]
     public var gatherCoverageData: Bool
+    public var commandLineArguments: [String: Bool]
```
Comment (yonaskolb) [inline Sources/ProjectSpec/Target.swift]: Would anyone want to specify disabled arguments? In other words is there a need to make this a dictionary as opposed to an array?

**Comment by rahul-malik** [comment]
@yonaskolb - should i revert then? i'm assuming `xcproj 1.6.0` isn't coming until the weekend

File: Sources/ProjectSpec/Target.swift (hunk 1 above)
Comment (yonaskolb) [reply Sources/ProjectSpec/Target.swift]: commandLineArguments or just arguments? What does the Xcode interface use? (on my phone at the moment)

File: Docs/ProjectSpec.md [hunk 2]
```diff
@@ -354,6 +355,9 @@ targets
         - Staging
         - Production
       gatherCoverageData: true
+      commandLineArguments:
```
Comment (yonaskolb) [inline Docs/ProjectSpec.md]: Most arguments will use “-myArgument” format, so its good to show quotes are needed

Comment (rahul-malik) [reply Docs/ProjectSpec.md]: 👍  I can update with quotes

File: Sources/ProjectSpec/Target.swift (hunk 1 above)
Comment (rahul-malik) [reply Sources/ProjectSpec/Target.swift]: - Disabled arguments are actually useful if you have different modes your program can run in but their behavior is conflicting. The advantage is that you could define all the configurations in the schema and let developers toggle them on/off during development
- The element in `xcscheme` is `CommandLineArguments` so it's based off that. The Xcode UI has a more verbose `Arguments passed on launch`

File: Docs/ProjectSpec.md (hunk 2 above)
Comment (rahul-malik) [reply Docs/ProjectSpec.md]: PR here: https://github.com/yonaskolb/XcodeGen/pull/176

File: Sources/ProjectSpec/Target.swift (hunk 1 above)
Comment (yonaskolb) [reply Sources/ProjectSpec/Target.swift]: Perhaps `launchArguments`? What do you think?

Comment (rahul-malik) [reply Sources/ProjectSpec/Target.swift]: I can submit an update for launch arguments

Comment (yonaskolb) [reply Sources/ProjectSpec/Target.swift]: Ok cool

**Comment by yonaskolb** [comment]
We’ll leave it for now, but a versioned release will need to be referenced before an xcodegen release is made

**Pull request merged**